src/lib/formats/sdf_dsk.cpp

```diff
@@ -0,0 +1,202 @@
+// copyright-holders:Wilbert Pol, tim lindner
+/*********************************************************************
+
+    formats/sdf_dsk.h
+
+    SDF disk images. Format created by Darren Atkinson for use with
+    his CoCoSDC floppy disk emulator.
+    
+    http://cocosdc.blogspot.com/p/sd-card-socket-sd-card-socket-is-push.html
+
+*********************************************************************/
+
+#include <assert.h>
+
+#include "sdf_dsk.h"
+
+
+sdf_format::sdf_format()
+{
+}
+
+
+const char *sdf_format::name() const
+{
+	return "sdf";
+}
+
+
+const char *sdf_format::description() const
+{
+	return "SDF disk image";
+}
+
+
+const char *sdf_format::extensions() const
+{
+	return "sdf";
+}
+
+
+int sdf_format::identify(io_generic *io, uint32_t form_factor)
+{
+	const int header_size = 512;
+	uint8_t header[header_size];
+
+	uint64_t size = io_generic_size(io);
+
+	if (size < header_size)
+	{
+		return 0;
+	}
+	
+	io_generic_read(io, header, 0, header_size);
+
+	int tracks = header[4];
+	int track_size = 256 + 6250 + 150; // header, track, padding
+	int heads = header[5];
+
+	// Check magic bytes
+	if (header[0] != 'S' && header[1] != 'D' && header[2] != 'F' && header[3] != '1')
+	{
+		return 0;
+	}
+
+	// Check heads
+	if (heads != 1 && heads != 2)
+	{
+		return 0;
+	}
+	
+	if (size == header_size + heads * tracks * track_size)
+	{
+		return 100;
+	}
+
+	return 0;
+}
+
+
+bool sdf_format::load(io_generic *io, uint32_t form_factor, floppy_image *image)
+{
+	const int header_size = 512;
+	uint8_t header[header_size];
+
+	io_generic_read(io, header, 0, header_size);
+
+	const int tracks = header[4];
+	const int track_size = 256 + 6250 + 150; // header, track, padding
+	const int heads = header[5];
+
+	if (heads == 2)
+	{
+		image->set_variant(floppy_image::DSDD);
+	}
+	else
+	{
+		image->set_variant(floppy_image::SSDD);
+	}
+
+	for (int track = 0; track < tracks; track++)
+	{
+		for (int head = 0; head < heads; head++)
+		{
+			std::vector<uint8_t> track_data(track_size);
+			std::vector<uint32_t> raw_track_data;
+			int iam_location = -1;
+			int idam_location[32];
+			int dam_location[32];
+
+			// Read track
+			io_generic_read(io, &track_data[0], header_size + ( heads * track + head ) * track_size, track_size);
+			
+			int sector_count = track_data[0];
+			
+			if (sector_count > 31) return false;
+			
+			// Transfer IDAM and DAM locations to table
+			
+			for (int i = 0; i < 32; i++)
+			{
+				if (i < sector_count )
+				{
+					idam_location[i] = ((track_data[ 8 * (i+1) + 1] << 8 | track_data[ 8 * (i+1)]) & 0x3FFF) - 4;
+					dam_location[i] = ((track_data[ 8 * (i+1) + 1 + 2] << 8 | track_data[ 8 * (i+1) + 2]) & 0x3FFF) - 4;
+					
+					if (idam_location[i] > 6250 + 256) return false;
+					if (dam_location[i] > 6250 + 256) return false;
+				}
+				else
+				{
+					idam_location[i] = -1;
+					dam_location[i] = -1;
+				}
+			}
+
+			// Find IAM location
+			for(int i = idam_location[0] - 1; i >= 256 + 3; i--)
```

Comment:
Spacing is inconsistent - you do `for (...` above, but here you've done `for(...` (missing space).